src/main/java/com/opentable/metrics/FloatingPointHistogram.java

```diff
@@ -0,0 +1,132 @@
+import com.codahale.metrics.Snapshot;
+
+/**
+ * This class extends {@link Histogram} to include floating-point {@link #update} functions. Because the underlying
+ * {@link Reservoir}s store only {@code long}s, this class also takes a {@link #scale} by which floating-point
+ * values are <em>multiplied</em> before being rounded and stored as {@code long}s. The {@link Snapshot} returned
+ * wraps the underlying {@link Histogram#getSnapshot()} return value with an implementation that
+ * &ldquo;un-scales&rdquo; the returned values by <em>dividing</em> by the {@link #scale}.
+ *
+ * <p>
+ * This is essentially just the ol' &ldquo;convert to cents&rdquo; trick, but you may choose the magnitude of the
+ * {@link #scale} based on the expected range of floating-point values stored as well as the desired precision you
+ * wish to preserve.
+ *
+ * <p>
+ * You may still of course still call {@link #update(int)} or {@link #update(long)}; behavior in this case is
+ * undefined.
+ *
+ * <p>
+ * cf. <a href="https://github.com/dropwizard/metrics/issues/863">this issue</a>
+ */
+public class FloatingPointHistogram extends Histogram {
+    private final double scale;
+
+    public FloatingPointHistogram(final Reservoir reservoir, final double scale) {
+        super(reservoir);
+        this.scale = scale;
+    }
+
+    public FloatingPointHistogram(final Reservoir reservoir) {
+        this(reservoir, 1);
+    }
+
+    public void update(final float value) {
+        update((double) value);
+    }
+
+    public void update(final double value) {
+        update(Math.round(scale * value));
+    }
+
+    @Override
+    public Snapshot getSnapshot() {
+        return new ScaledSnapshot(super.getSnapshot());
+    }
+
+    private class ScaledSnapshot extends Snapshot {
+        private final Snapshot snap;
+
+        ScaledSnapshot(final Snapshot snap) {
+            this.snap = snap;
+        }
+
+        private double scaled(final double value) {
+            return value / scale;
+        }
+
+        private long scaled(final long value) {
+            return Math.round(scaled((double) value));
+        }
+
+        @Override
+        public double getValue(final double quantile) {
+            return scaled(snap.getValue(quantile));
+        }
+
+        @Override
+        public long[] getValues() {
+            return Arrays
+                    .stream(snap.getValues())
+                    .map(this::scaled)
+                    .toArray();
+        }
+
+        @Override
+        public int size() {
+            return snap.size();
+        }
+
+        @Override
+        public long getMax() {
+            return scaled(snap.getMax());
+        }
+
+        @Override
+        public double getMean() {
+            return scaled(snap.getMean());
+        }
+
+        @Override
+        public long getMin() {
+            return scaled(snap.getMin());
+        }
+
+        @Override
+        public double getStdDev() {
```

Comment:
I buy the rest of it, but is scaling the standard deviation in this way safe?

Reply:
Hehe, I was wondering if you'd catch this.  I'm not sure.  Let me do some fiddling and see how much it might not make sense.

Reply:
So in fact, when you scale a random variable by a constant (as we're doing here), the variance is scaled by the square of that constant.  Therefore, the standard deviation is scaled by that constant as well.

I re-derived that myself, and then found it listed [here as well][1].

So this is actually correct!

[1]: https://en.wikipedia.org/wiki/Variance#Basic_properties